src/events/guildMemberUpdate.js

```diff
@@ -1,29 +1,101 @@
+    if (!oldmemberFlags.CompletedOnboarding && !newmemberFlags.CompletedOnboarding) return
     /**
      *  -------------------------------------------------------
      *  WELCOMER MODULE
      *  -------------------------------------------------------
      */
-     if (configs.get(`WELCOMER_MODULE`).value) {
-        // If the user still needs to complete the discord membership gate for this guild
-        if (newMember.pending) return
-        /**
+    if (configs.get(`WELCOMER_MODULE`).value) {
+        if (!newMember.pending) {
+            /**
          *  -------------------------------------------------------
          *  WELCOMER'S AUTOROLE MODULE
          *  -------------------------------------------------------
          */
-        //  Skip role assignment if no roles are registered
-        const welcomerRolesList = configs.get(`WELCOMER_ROLES`)
-        if (welcomerRolesList.value.length <= 0) return
-        for (let i=0; i<welcomerRolesList.value.length; i++) {
-            const roleId = welcomerRolesList.value[i]
-            //  Handle if role cannot be found due to deleted/invalid
-            if (!guild.roles.cache.has(roleId)) continue
-            newMember.roles.add(roleId)
+            //  Skip role assignment if no roles are registered
+            const welcomerRolesList = configs.get(`WELCOMER_ROLES`)
+            if (welcomerRolesList.value.length <= 0) return
+            for (let i = 0; i < welcomerRolesList.value.length; i++) {
+                const roleId = welcomerRolesList.value[i]
+                //  Handle if role cannot be found due to deleted/invalid
+                if (!guild.roles.cache.has(roleId)) continue
+                newMember.roles.add(roleId)
+            }
         }
+
+        if (!configs.get(`WELCOMER_ONBOARDWAIT`).value) return
+        //  Prepare welcomer target channel
+        let getTargetWelcomerChannel = configs.get(`WELCOMER_CHANNEL`).value
+        //  Prepare welcomer banner img
+        let renderedBanner = await new Banner(newMember, client).build()
+        //  Prepare greeting text
+        let welcomerText = configs.get(`WELCOMER_TEXT`)
+        let getWelcomerText = parseWelcomerText(welcomerText.value, guild, newMember)
+        //  Attempt to DM the joined user if guild's owner hasn't setup the welcomer channel yet
+        if (!getTargetWelcomerChannel) {
+            client.responseLibs(newMember, true)
+                .send(`__**A letter from ${guild.name}.**__\n` + getWelcomerText, {
+                    simplified: true,
+                    prebuffer: true,
+                    image: renderedBanner
+                })
+                .catch(() => client.logger.warn(`${instance} failed to send the requested welcomer message due to there was no welcomer channel and the user's dm were locked.`))
+        } else {
+            /**
+             * Handle multiple channels
+             */
+            if (configs.get(`WELCOMER_ADDITIONAL_CHANNELS`)) {
```

Comment:
` if (configs.get('WELCOMER_ADDITIONAL_CHANNELS')) {`
you missed the `.value` property here; otherwise, this condition will always stay true lol